### PR TITLE
Hydrogen ore processing fix

### LINKED ISSUE #8980: Hydrogen ore doesn't appear to be processed.
Round-ID: b7o-bYaO

After mining a lot of hydrogen, none of it was processed. Upon investigation, adding hydrogen to the smelter didn't increase any ore counter in the smelter, suggesting that hydrogen wasn't being processed.

### PR DESCRIPTION
**Pull request by aleksix**
Hydrogen ore is now properly processed in the mining furnace. Will most likely contribute to the Drill Technician powercreep, as they generate a lot of hydrogen and hydrogen costs were unaffected by the fix.

Fixes #8980

### CODE REVIEW

File: html/changelogs/hydrogen_processing-aleksix.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - bugfix: "Hyrogen ore is now processed properly."
```
Comment (Geevies) [inline html/changelogs/hydrogen_processing-aleksix.yml:41]:
```suggestion
  - bugfix: "Hydrogen ore is now processed properly."
```

**Comment by aleksix** [comment]
!bugfix

**Comment by aleksix** [comment]
!review

**Comment by aleksix** [comment]
> 
> 
> wow, i was looking everywhere but here when looking for a fix

Yeah, I didn't expect that one either. There is a collision on some names, but since the names are assigned to completely different classes, there shouldn't be an issue.

**Pull request merged**